src/utils/internal/parseGraphQLRequest.ts

```diff
@@ -86,24 +86,43 @@ function extractMultipartVariables<VariablesType extends GraphQLVariables>(
   return operations.variables
 }
 
+/**
+ * Check if the given GraphQL query string is potentially valid
+ * without parsing it. This returns false fast when analyzing
+ * GET/POST requests with the "query" parameter/body property
+ * which are not a GraphQL request.
+ */
+function maybeGraphQLQuery(query?: string | null): query is string {
+  const normalizedQuery = query?.toLowerCase().trim() || ''
+
+  return (
+    normalizedQuery.startsWith('query') ||
```

Comment:
This will return `false` fast for GraphQL queries with comments. But I don't think it's allowed for the query string to contain comments, I'd expect GraphQL clients should strip them. 

That'd be nice to be researched.